Code/LauncherUnified/Game.cpp

```diff
@@ -27,4 +27,13 @@ namespace O3DELauncher
     {
         return "client";
     }
+
+    bool IsHeadless()
+    {
+        #if LY_HEADLESS
+        return true;
+        #else
+        return false;
+        #endif
+    }
```

Comment:
The GameLauncher target would never be headless.

So this should just be return false.
```suggestion
    bool IsHeadless()
    {
        return false;
    }
```

Reply:
Also shouldn't the Server.cpp and Unified.cpp implement this function?

Also with the implementation, can pass the `#define` directly into a boolean variable.
That can account to for the case where the developer explicitly sets O3DE_HEADLESS_SERVER to 0 as well as it not being defined.
```c++
bool IsHeadless()
{
#if defined(O3DE_HEADLESS_SERVER )
    return bool{ O3DE_HEADLESS_SERVER };
#else
    return false;
#endif
}
```

Reply:
I used `#if O3DE_HEADLESS_SERVER` instead of `#if defined` because the `#if` evaluates if `O3DE_HEADLESS_SERVER` is sete to 0 or it is not defined at all. Needless to say, this function is a left-over and not used at all